### PR TITLE
Redis Cache users/userID/tasks and Perf Test Fixes

### LINKED ISSUE #134: [Task] Create Redis cache for users/{userID}/tasks
**Is your feature request related to a problem? Please describe.**
We can optimize how quickly our app can deal with multiple identical requests by making a Redis cache that will temporarily hold data from previous requests, preventing need to access the SQL server multiple times within a small period of time. 

**Describe the solution you'd like**
A Redis cache that holds Read requests for 15 seconds for users/{userID}/tasks, and will divert future read requests for those 15 seconds

**Describe alternatives you've considered**
MongoDB, ArangoDB, and other virtual Databases would also fit here.

### PR DESCRIPTION
**Pull request by kieranmiguel**
-Added Redis cache for users/userID/tasks that expires after 20 seconds, and reroutes subsequent requests within those 20 seconds to the Cache 

-Redis cache is initialized along with SQL sever, and attempts to connect with try/catch implementation and logging.
-When a get request is made, all tasks are loaded into a JSON object, and then loaded into the redis server as users:user_id:tasks:all, immediately after which, an expire tag is added to the cache, to expire users:user_id:tasks:all after 20 seconds. If another get request is made within those 20 seconds, the request will get from the redis cache instead of the SQL server, reducing lag. 

-Fixes for Performance Script

### CODE REVIEW

File: functions/HttpTriggerAPIUsersIdTask/__init__.py [hunk 1]
```diff
@@ -33,6 +39,16 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
         return func.HttpResponse(status_code=500)
     logging.debug("Connected to DB successfuly!")
 
+    #Redis Sever
+    try:
+        
+        rDB = redis.Redis(host='nsc-redis-dev-usw2-thursday.redis.cache.windows.net', port='6380', db=0, password='${{ secrets.ENV_REDIS_KEY }}', ssl=True) 
+        rDB.ping()
```
Comment (chonger878) [inline functions/HttpTriggerAPIUsersIdTask/__init__.py:46]: Hm, I am guessing this lets user know somehow that they were/were not able to get onto the Redis DB?

**Review by chonger878 — APPROVED** [review]
Wow, I forgot that the further you do down the url, the more complex the code has to be.

**Review by jaguiniga — APPROVED** [review]
Looks like you added secrets for keep the password secure. Also the correct Host.

File: functions/HttpTriggerAPIUsersIdTask/__init__.py [hunk 2]
```diff
@@ -33,6 +39,16 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
         return func.HttpResponse(status_code=500)
     logging.debug("Connected to DB successfuly!")
 
+    #Redis Sever
+    try:
+        
+        rDB = redis.Redis(host='nsc-redis-dev-usw2-thursday.redis.cache.windows.net', port='6380', db=0, password='${{ secrets.ENV_REDIS_KEY }}', ssl=True)
```
Comment (toddysm) [inline functions/HttpTriggerAPIUsersIdTask/__init__.py:45]: What if the Redis server host changes?

File: functions/HttpTriggerAPIUsersIdTask/__init__.py [hunk 3]
```diff
@@ -86,11 +105,24 @@ def get_user_tasks(conn, userId):
         tasks = []
         columns = [column[0] for column in cursor.description]
 
+    if(rDB.get('UsersIdTasks')):
+        unpacked_tasks = json.loads(rDB.get('tasks'))
```
Comment (toddysm) [inline functions/HttpTriggerAPIUsersIdTask/__init__.py:109]: The tasks for which user are you pulling here?

File: functions/HttpTriggerAPIUsersIdTask/__init__.py (hunk 1 above)
Comment (kieranmiguel) [reply functions/HttpTriggerAPIUsersIdTask/__init__.py:46]: Ill add this in the next sprint...

**Pull request merged**